pyasn1/codec/ber/decoder.py

```diff
@@ -16,35 +19,71 @@
 from pyasn1.type import univ
 from pyasn1.type import useful
 
-__all__ = ['decode']
+
+__all__ = ['decode', 'decodeStream']
 
 LOG = debug.registerLoggee(__name__, flags=debug.DEBUG_DECODER)
 
 noValue = base.noValue
 
 
+def asStream(substrate):
+    # TODO: Apply sparingly or remove
+    # :type substrate: Union[bytes, BytesIO, BufferedReader, OctetString]
```

Comment:
We should probably add annotations everywhere uniformly...

Reply:
Sure, these were mostly comments for my grasping of code. I will happily conform to any style guide you have.

Reply:
There is no official style guide. The original code dates back to Python 1.5 times, so I am trying to update the style and idioms where it does not break Python 2.4 compatibility (for current master). However the next step is to drop 2.4, 2.5 and possibly 2.6 support - that would let us code in contemporary Python. \o/